HospitalSystem.ConsoleUI/PatientUI.cs

```diff
@@ -983,8 +988,75 @@ public void StartDoctorSearch()
 
         CreateCheckup(filteredDoctors[selectedIndex]);
         
+    }
 
+    public void ManageNotifications()
+    {
+        System.Console.WriteLine(@"
+            Options:
+            s - show notifications
+            w - set when to remind
+            ");
+        string sortOption = ReadSanitizedLine().Trim();
+        if (sortOption == "s")
+        {
+            ShowNotifications();
+        }
+        else if (sortOption == "w")
+        {
+            SetNotificationSettings();
+        }
     }
 
-    
-}
\ No newline at end of file
+    public void ShowNotifications()
+    {
+        int notificationCount = 0;
+        foreach (Prescription prescription in _loggedInPatient.MedicalRecord.Prescriptions)
+        {
+            DateTime ?whenToTake = _hospital.PatientService.WhenToTakeMedicine(prescription,_loggedInPatient);
+            if (whenToTake is not null)
+            {
+                notificationCount += 1;
+                string meal = "";
+                switch(prescription.BestTaken)
+                {
+                    case MedicationBestTaken.AFTER_MEAL:
+                    meal = "after meal";
+                    break;
+                    case MedicationBestTaken.BEFORE_MEAL:
+                    meal = "before meal";
+                    break;
+                    case MedicationBestTaken.ANY_TIME:
+                    meal = "any time";
+                    break;
+                    case MedicationBestTaken.WITH_MEAL:
+                    meal = "with meal";
+                    break;
+                }
+                Console.WriteLine("Take "+prescription.Medication.Name+" at "+ whenToTake?.ToString("HH:mm")+" best taken "+meal);
+            }
+        }
+        if (notificationCount==0)
+        {
+            Console.WriteLine("No notifications.");
+        }
+    }
+    public void SetNotificationSettings()
+    {
+        int numberOfMinutes;
+        Console.WriteLine("Please enter how many minutes before perscription should be considered (min 5, max 300): ");
+        try
+        {
+            numberOfMinutes = ReadInt(5, 300, "Number out of bounds!", "Number not recognized!");
+        }
+        catch (InvalidInputException e)
+        {
+            System.Console.Write(e.Message + " Aborting...");
+            throw new QuitToMainMenuException("Wrong input");
```

Comment:
Wrong exception type. QTMM should be used for user driven quitting

Reply:
I'm aware of that, i'll fix it later.